pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1585,13 +1586,26 @@ public void updateSchema(Schema schema, boolean reload, boolean forceTableSchema
     }
 
     updateSchema(schema, oldSchema, forceTableSchemaUpdate);
-
-    if (reload) {
-      LOGGER.info("Reloading tables with name: {}", schemaName);
-      List<String> tableNamesWithType = getExistingTableNamesWithType(schemaName, null);
+    List<String> tableNamesWithType;
+    try {
+      tableNamesWithType = getExistingTableNamesWithType(schemaName, null);
+      if (reload) {
+        LOGGER.info("Reloading tables with name: {}", schemaName);
+        for (String tableNameWithType : tableNamesWithType) {
+          reloadAllSegments(tableNameWithType, false, null);
+        }
+      }
+      // Send schema refresh message to all tables that use this schema
       for (String tableNameWithType : tableNamesWithType) {
-        reloadAllSegments(tableNameWithType, false, null);
+        LOGGER.info("Sending updated schema message for table: {}", tableNameWithType);
+        sendTableSchemaRefreshMessage(tableNameWithType);
```

Comment:
We don't need to send this if reload is enabled. Reload will also fetch new table config and schema